picard/file.py

```diff
@@ -165,7 +165,17 @@ def _load_check(self, filename):
         if self.tagger.stopping:
             log.debug("File not loaded because %s is stopping: %r", PICARD_APP_NAME, self.filename)
             return None
-        return self._load(filename)
+        try:
+            # Try loading based on extension first
+            ret = self._load(filename)
```

Comment:
After those changes, I guess we could just `return self._load(filename)` and get rid of `ret`.